src/fsharp/FSharp.Core/prim-types.fs

```diff
@@ -2336,7 +2357,22 @@ namespace Microsoft.FSharp.Core
             | 'o' -> parseOctalUInt64  (s.Substring(p))
             | _ -> UInt64.Parse(s.Substring(p), NumberStyles.AllowLeadingSign, CultureInfo.InvariantCulture) 
 
+        let inline ParseByte (s:string)       = (# "conv.ovf.u1" (ParseUInt32 s) : byte #)
+
+        let inline ParseSByte (s:string)      = (# "conv.ovf.i1" (ParseInt32 s)  : sbyte #)
+
+        let inline ParseInt16 (s:string)      = (# "conv.ovf.i2" (ParseInt32 s)  : int16 #)
+
+        let inline ParseUInt16 (s:string)     = (# "conv.ovf.u2" (ParseUInt32 s) : uint16 #)
+
+        let inline ParseIntPtr (s:string)  = (# "conv.ovf.i"  (ParseInt64 s)  : nativeint #)
+
+        let inline ParseUIntPtr (s:string) = (# "conv.ovf.u"  (ParseInt64 s)  : unativeint #)
```

Comment:
Should these just use the .NET methods?

Reply:
For the purposes of this PR let's ignore this, as the code is just moved around to put the dynamic invocation tables together